### PR TITLE
Dev env script and update for Ubuntu 18.04/ROS Melodic

### LINKED ISSUE #751: Clarify simulator/ROS and Linux versions
[Linux · PX4 Developer Guide](http://dev.px4.io/en/setup/dev_env_linux.html) has a number of scripts that were developed on Ubuntu 16.04. I'd like to retest on Ubuntu 18.04 and provide updated/recommended environment.

Three main scripts:
- [ubuntu_sim_nuttx.sh](https://raw.githubusercontent.com/PX4/Devguide/master/build_scripts/ubuntu_sim_nuttx.sh) = nuttx and jmavsim. 
  - Testing shows fine on Ubuntu 18.04
- [ubuntu_sim.sh](https://raw.githubusercontent.com/PX4/Devguide/master/build_scripts/ubuntu_sim.sh) += Gazebo 9. 
  - Testing shows works fine on Ubuntu 18.04.
- [ubuntu_sim_ros_gazebo.sh](https://raw.githubusercontent.com/PX4/Devguide/master/build_scripts/ubuntu_sim_ros_gazebo.sh) = ROS Kinetic with Gazebo 7 default.
  - Fails, log below.

### PR DESCRIPTION
**Pull request by hamishwillee**
Fixes #751

This is required because PX4 updated to ROS melodic, which only runs on 18.04. 
This will require a little more work, but would be good to get the new script reviewed before we proceed.

### CODE REVIEW

File: build_scripts/ubuntu_sim_ros_melodic.sh [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+sudo apt-get install python-catkin-tools python-rosinstall-generator -y
+
+## Create catkin workspace
+mkdir -p ~/catkin_ws/src
+cd ~/catkin_ws
+catkin init
+wstool init src
+
+
+## Install MAVLink
```
Comment (hamishwillee) [inline build_scripts/ubuntu_sim_ros_melodic.sh:57]: @TSC21 Following our discussion in https://github.com/PX4/Devguide/issues/751 this is a new script for ROS melodic installation on Ubuntu 18.04. It is based on http://wiki.ros.org/melodic/Installation/Ubuntu and https://github.com/mavlink/mavros/tree/master/mavros#required-dependencies

Can you minimally check the mavros part is correct (it mentions ROS Kinetic, so feels "uncomfortable"). 
The full build log is [here](https://gist.github.com/hamishwillee/2b32eb11d7cfad4efa1d92b2b8c8b6d6) - it looks good except for an error in RTPS build. 
I suspect that is a source rather than setup problem.

Basically I'm hoping you can tell me that this script is indeed OK, and then I can tidy up rest of docs.

**Comment by hamishwillee** [comment]
PS When you say we have moved to ROS Melodic, what does that mean for PX4 and ROS Kinetic? Ie if you're on PX4 1.8.2 should you be using ROS Kinetic, what about PX4 1.8.0 and earlier?

File: build_scripts/ubuntu_sim_ros_melodic.sh [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+### Get source (upstream - released)
+rosinstall_generator --upstream mavros | tee -a /tmp/mavros.rosinstall
+
+### Setup workspace & install deps
+wstool merge -t src /tmp/mavros.rosinstall
+wstool update -t src
+if ! rosdep install --from-paths src --ignore-src -y; then
+    # (Use echo to trim leading/trailing whitespaces from the unsupported OS name
+    unsupported_os=$(echo $(rosdep db 2>&1| grep Unsupported | awk -F: '{print $2}'))
+    rosdep install --from-paths src --ignore-src --rosdistro melodic -y --os ubuntu:bionic
```
Comment (hamishwillee) [inline build_scripts/ubuntu_sim_ros_melodic.sh:71]: FYI I updated this line to refer to melodic/bionic.

File: build_scripts/ubuntu_sim_ros_melodic.sh [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+fi
+
+if [[ ! -z $unsupported_os ]]; then
```
Comment (hamishwillee) [inline build_scripts/ubuntu_sim_ros_melodic.sh:74]: FYI, this tests for version match on OS based on what ROS itself says. Do you think we should kill installation at this point rather than allowing it to continue?

**Comment by hamishwillee** [comment]
@TSC21 Would appreciate your sanity check.

File: build_scripts/ubuntu_sim_ros_melodic.sh (hunk 1 above)
Comment (TSC21) [reply build_scripts/ubuntu_sim_ros_melodic.sh:57]: We should actually be installing Fast-RTPS 1.7.1 instead of 1.5.0.

File: build_scripts/ubuntu_sim_ros_melodic.sh (hunk 3 above)
Comment (TSC21) [reply build_scripts/ubuntu_sim_ros_melodic.sh:74]: It should be killed.

**Review by TSC21 — APPROVED** [review]

File: build_scripts/ubuntu_sim_ros_melodic.sh (hunk 1 above)
Comment (hamishwillee) [reply build_scripts/ubuntu_sim_ros_melodic.sh:57]: OK. I've updated (and tested) that script too.

File: build_scripts/ubuntu_sim_ros_melodic.sh (hunk 3 above)
Comment (hamishwillee) [reply build_scripts/ubuntu_sim_ros_melodic.sh:74]: Tried this, but first attempt did not kill this as expected. I'm going to merge this now and look at it again later.

Comment (hamishwillee) [reply build_scripts/ubuntu_sim_ros_melodic.sh:74]: @TSC21 I think that by the time you get to this point on Ubuntu 16.04 things have already failed pretty badly. I'm thinking perhaps we exit based on a xenial check in the first line?

Comment (TSC21) [reply build_scripts/ubuntu_sim_ros_melodic.sh:74]: I am ok with that

Comment (hamishwillee) [reply build_scripts/ubuntu_sim_ros_melodic.sh:74]: Great. Done.
